cpp/CMakeLists.txt

```diff
@@ -3,14 +3,23 @@ cmake_minimum_required(VERSION 3.11...3.21)
 project(XoscarCollective)
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
-if(NOT DEFINED LIBRARY_OUTPUT_DIRECTORY)
-  set(LIBRARY_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/python/xoscar/collective/rendezvous)
+if(NOT DEFINED LIBRARY_OUTPUT_DIRECTORY_STORE)
+  set(LIBRARY_OUTPUT_DIRECTORY_STORE ${CMAKE_SOURCE_DIR}/python/xoscar/collective/rendezvous)
+endif()
+
+if(NOT DEFINED LIBRARY_OUTPUT_DIRECTORY_GLOO)
+  set(LIBRARY_OUTPUT_DIRECTORY_GLOO ${CMAKE_SOURCE_DIR}/python/xoscar/collective/gloo)
```

Comment:
Since two parts of cpp codes have been merged, we only need on output directory, right?